persist/fs/seek.go

```diff
@@ -181,6 +181,13 @@ func (s *seeker) Open(namespace ts.ID, shard uint32, blockStart time.Time) error
 		return err
 	}
 
+	if !s.keepUnreadBuf {
```

Comment:
The primary use case is:
```
1. Set the unread buffer externally for reuse across multiple seekers
2. Call `Open` for each seeker
3. In readIndex, the buffer is nil'ed out after the index is read
```

If we instead nil out the buffer early on in `Open`, it means the buffer will always be reallocated instead of being reused across seekers, which incurs significant memory/GC overhead.